src/H5FDdevelop.h

```diff
@@ -196,12 +196,12 @@ typedef struct H5FD_class_t {
                           size_t sizes[], void *bufs[]);
     herr_t (*write_vector)(H5FD_t *file, hid_t dxpl, uint32_t count, H5FD_mem_t types[], haddr_t addrs[],
                            size_t sizes[], const void *bufs[]);
-    herr_t (*read_selection)(H5FD_t *file, H5FD_mem_t type, hid_t dxpl_id, size_t count, hid_t mem_spaces[],
```

Comment:
I understand we want to keep things consistent but if we're going to change the API to keep the count parameter the same type we should change it to size_t. I think we should just lave it alone for now though.

Reply:
So I will keep <count > as size_t for read_selection() and write_selection()? But <count> as uint32_t for write_vector and read_vector?

Reply:
Yeah I don't think we should change it now, it's already part of the API

Reply:
Just want to double-check:
<count> as size_t will be used for all APIs and the related intermediate routines:
H5FDread/write_selection(), H5FDread/write_vector_from_selection(), H5FDread/write_from_selection()

Reply:
correction to previous comment: "count" as size_t

Reply:
Don't change any APIs right now, thanks, just revert the type change to the H5FD_class_t struct